File: agri-ecommerce-backend/src/main/java/com/agri/ecommerce/service/CartService.java
package com.agri.ecommerce.service;

import com.agri.ecommerce.dto.request.cart.AddCartItemRequest;
import com.agri.ecommerce.dto.request.cart.UpdateCartItemRequest;
import com.agri.ecommerce.dto.response.cart.CartResponse;

public interface CartService {

    CartResponse getCart(Long userId);

    CartResponse addItem(Long userId, AddCartItemRequest request);

    CartResponse updateItem(Long userId, Long itemId, UpdateCartItemRequest request);

    CartResponse removeItem(Long userId, Long itemId);

    CartResponse clearCart(Long userId);
}

File: agri-ecommerce-frontend/src/services/cart.service.js
import axiosClient from "@/lib/axios-client";

const unwrapApiData = (response) => response?.data ?? response;

export const cartService = {
  getCart: async () => {
    const response = await axiosClient.get("/customer/cart");
    return unwrapApiData(response);
  },

  addItem: async (payload) => {
    const response = await axiosClient.post("/customer/cart/items", payload);
    return unwrapApiData(response);
  },

  updateItem: async (itemId, payload) => {
    const response = await axiosClient.put(
      `/customer/cart/items/${itemId}`,
      payload
    );
    return unwrapApiData(response);
  },

  removeItem: async (itemId) => {
    const response = await axiosClient.delete(`/customer/cart/items/${itemId}`);
    return unwrapApiData(response);
  },

  clearCart: async () => {
    const response = await axiosClient.delete("/customer/cart");
    return unwrapApiData(response);
  },
};
